pkg/provider/vault/auth.go

```diff
@@ -179,10 +179,10 @@ func revokeTokenIfValid(ctx context.Context, client util.Client) error {
 	return nil
 }
 
-func (c *client) useAuthNamespace(_ context.Context) func() {
+func (c *client) useAuthNamespace(_ context.Context, spec *esv1beta1.VaultProvider) func() {
 	ns := ""
-	if c.store.Namespace != nil {
-		ns = *c.store.Namespace
+	if spec != nil && spec.Namespace != nil {
```

Comment:
@moolen Okay, I _think_ this was restoring the wrong namespace 🤔 It was restoring store.Namespace instead of the provider namespace.

This is, of course, only a problem if auth.Namespace was set. Let's see if that was the case here.

Reply:
I don't think that would change anything. `c.store` is actually the same `*esv1beta1.VaultProvider` https://github.com/external-secrets/external-secrets/blob/main/pkg/provider/vault/client.go#L42

Reply:
I believe the actual problem is that the current `useAuthNamespace` is a nil operation whenever there isn't this flag, and it is just set with a `defer()` on https://github.com/external-secrets/external-secrets/blob/main/pkg/provider/vault/auth.go#L45-L48. Meaning any namespace-based logins will fail.

Reply:
Yep. You were looking at an old commit here? 🤔 In the new one, I fixed the defer to return the set namespace if any was set before. :) 👍